src/brokers/broker-details/BrokerDetails.container.tsx

```diff
@@ -153,15 +140,40 @@ export const BrokerDetailsPage: FC = () => {
   const { ns: namespace, name } = useParams<{ ns?: string; name?: string }>();
 
   const { brokerCr, isLoading, error } = useGetBrokerCR(name, namespace);
+  const { t } = useTranslation();
+
+  if (isLoading) {
+    return (
+      <Bullseye>
+        <Spinner size="xl" />
+      </Bullseye>
+    );
+  }
+
+  if (error) {
+    return (
+      <EmptyState>
+        <EmptyStateHeader
+          titleText={t('Error while fetching broker-details')}
+          icon={<EmptyStateIcon icon={ErrorCircleOIcon} />}
+          headingLevel="h4"
+        />
+        <EmptyStateBody>
+          {t(
+            'Could not fetch broker details. An unexpected error occurred while loading the broker details. Please try again later.',
+          )}
+        </EmptyStateBody>
+      </EmptyState>
+    );
+  }
+
```

Comment:
This loading and error handling being the same for the Broker details page and the Address details page, I think it would be good to factorize things a bit.

Can you extract these two components in the share components somewhere please and then reference them in both the broker details view and the address details view please?